update

qPCR4vir opened the pull request:
Sorry, peaple are trying to use your fork, but it is outdated and don't work any more.

https://stackoverflow.com/questions/45778686/setup-nana-with-cmake-in-clion

qPCR4vir committed:
pending with then mouse.

qPCR4vir committed:
fail to write in notepad!?

qPCR4vir committed:
minimal GUI test exec(1, [](){},1,&fm)

qPCR4vir committed:
undo automatic CLion additions

qPCR4vir committed:
fix

qPCR4vir committed:
int main(){} -> main.cpp  [skip ci]

qPCR4vir committed:
std::cout << "Hello main...\n";

qPCR4vir committed:
#include <nana/gui/wvl.hpp>

qPCR4vir committed:
nana form  fm.show()

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
<< std::endl;

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
std

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
exec();

qPCR4vir committed:
Update main.cpp

qPCR4vir committed:
Update MontiHall.cpp

qPCR4vir committed:
Update a_group_impl.cpp

qPCR4vir committed:
Update animate-bmp.cpp

qPCR4vir committed:
Update HelloWord.cpp

qPCR4vir committed:
Update background-effects.cpp

qPCR4vir committed:
Update categ.cpp

qPCR4vir committed:
Update decore.cpp

qPCR4vir committed:
Update example.button.cpp

qPCR4vir committed:
Update dock.cpp

qPCR4vir committed:
Update drag-button.cpp

qPCR4vir committed:
Update draw.cpp

qPCR4vir committed:
Update file_explorer.cpp

qPCR4vir committed:
Update example_menu.cpp

qPCR4vir committed:
Update example_listbox.cpp

qPCR4vir committed:
Update example_combox.cpp

qPCR4vir committed:
Update example.button.cpp

qPCR4vir committed:
automatic GUI tested in windows VS2015

qPCR4vir committed:
more

qPCR4vir committed:
fix main

qPCR4vir committed:
fix

qPCR4vir committed:
fix

qPCR4vir committed:
adding automatic GUI testing

qPCR4vir committed:
all demos adapted

qPCR4vir committed:
fix exec

qPCR4vir committed:
fix main

qPCR4vir committed:
"Press a char: !

cnjinhao committed:
added a new parameter for basic_event::emit method

see nana/develop branch(07b5b195455ea1db402619861737dabab7696f86)

cnjinhao committed:
Add a new example

cnjinhao committed:
Add a new example

cnjinhao committed:
Update stretch_image.cpp

A new commit of Nana has removed some conversion constructors of point/size/rectangle.

cnjinhao committed:
a workaround for VC2013

cnjinhao committed:
a fix

cnjinhao committed:
add -lstd++fs

cnjinhao committed:
option NANA_CMAKE_INCLUDE_EXPERIMENTAL_DEMOS off

cnjinhao committed:
remove helloworld_demo

cnjinhao committed:
adapte to v1.4

qPCR4vir committed:
using explicit namespaces fs

qPCR4vir committed:
FIX crash in VS consistently using filesysten_ext

qPCR4vir committed:
set in config...

qPCR4vir committed:
preparing build independently at nana build

qPCR4vir committed:
ignore *.VC.db

qPCR4vir committed:
finding nana lib

qPCR4vir committed:
finding nana lib in ../nana_lib/lib

qPCR4vir committed:
force libnana.a

qPCR4vir committed:
link exe flags, use dir/

qPCR4vir committed:
find_library works

qPCR4vir committed:
target_link_libraries(${test}) only

qPCR4vir committed:
restore nanalib

qPCR4vir committed:
"-l${nana_lib}"

qPCR4vir committed:
target_link_libraries(${test} ${nana_lib})

qPCR4vir committed:
return all options

qPCR4vir committed:
install DCMAKE_INSTALL_PREFIX=..

qPCR4vir committed:
install(DIRECTORY ${examples_dir} DESTINATION ${examples_dir} )

qPCR4vir committed:
add more options to find nana

qPCR4vir committed:
add more options to find nana

qPCR4vir committed:
Use my onw absolute install path (CLion workaround!)

qPCR4vir committed:
add travis to demos

qPCR4vir committed:
logo travis [ci skip]

qPCR4vir committed:
cleaning

qPCR4vir committed:
Merge remote-tracking branch 'remotes/origin/dev_nana_fs' into dev_nana_fs

qPCR4vir committed:
nana links? order?

qPCR4vir committed:
Merge branch 'develop_with_nana' into develop

Conflicts:
	Examples/folder_tree_nana.cpp
	Examples/folder_tree_std.cpp
	calculator.cpp
	widget_show.cpp
	widget_show2.cpp

qPCR4vir committed:
more places to find nana.lib

qPCR4vir committed:
test all in order

qPCR4vir committed:
works with:  branch=hotfix-1.4 https://github.com/cnjinhao/nana.git

qPCR4vir committed:
34 of 54 run

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
works with:  branch=master https://github.com/cnjinhao/nana.git

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
include map

some nana's headers have been eliminated the dependency of standard header.

cnjinhao committed:
Added notify_status

notify_status is introduced since 1.4.1

cnjinhao committed:
Added notify_status

cnjinhao committed:
Added notify_status

cnjinhao committed:
Added notify_status

qPCR4vir committed:
Travis usage of branchs in *nana* and *nana-demo*:

qPCR4vir committed:
Travis usage of branchs in *nana* and *nana-demo*:

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
use cnjinhao/nana/hotfix-1.4.1

qPCR4vir committed:
readme travis info- gcc version, branch correspondence

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
add .gitignore  .travis.yml and README.md to the solution

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into develop_with_nana

qPCR4vir committed:
link to nana/develop

qPCR4vir committed:
Merge branch 'develop_with_nana' into develop

qPCR4vir committed:
add license

qPCR4vir committed:
add license

qPCR4vir committed:
add license

qPCR4vir committed:
add license

qPCR4vir committed:
ignore vs, cmake and codelite temp

qPCR4vir committed:
https://github.com/cnjinhao/nana/issues/191#issuecomment-287226668

qPCR4vir committed:
vc2017

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
Merge branch 'hotfix-1.4' into develop_with_nana

qPCR4vir committed:
Merge branch 'develop_with_nana' into develop

qPCR4vir committed:
Merge branch 'hotfix-1.4_with_nana' into hotfix-1.4

qPCR4vir committed:
Merge branch 'develop'

# Conflicts:
#	.travis.yml
#	Examples/listbox_inline_widget.cpp
#	README.md

qPCR4vir committed:
clone -branch=hotfix-1.5 cnjinhao/nana

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into hotfix-1.5

qPCR4vir committed:
updating to hotfix-1.5

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into hotfix-1.5

cnjinhao committed:
this example requires 1.5.4

add the calculation of top line position in y-axis

cnjinhao committed:
this example requires 1.5.4

add the calculation of top line position in y-axis

qPCR4vir committed:
Merge pull request #12 from qPCR4vir/cnjinhao-patch-1

this example requires 1.5.4

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into develop_with_nana

# Conflicts:
#	.travis.yml

qPCR4vir committed:
add new example listbox-stl for VS2017

qPCR4vir committed:
add new example textbox_line_number for VS2017

qPCR4vir committed:
new solution for VS2017

qPCR4vir committed:
add new #ifdef NANA_AUTOMATIC_GUI_TESTING for a_group_impl

qPCR4vir committed:
#ifdef NANA_AUTOMATIC_GUI_TESTING

qPCR4vir committed:
calculator #ifdef NANA_AUTOMATIC_GUI_TESTING, and reduce wait for travis

qPCR4vir committed:
fix: NANA_AUTOMATIC_GUI_TESTING, and test faster for travis
by def, wait only 1 sec. By manual testing change that to 2,30 (2 sec between clicks and 30 at the end)

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into hotfix-1.5

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into develop_with_nana

qPCR4vir committed:
Merge branch 'develop_with_nana' into develop

qPCR4vir committed:
Merge branch 'hotfix-1.5'

# Conflicts:
#	.travis.yml

qPCR4vir committed:
use git clone --branch=hotfix-1.5.4

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into hotfix-1.5

qPCR4vir committed:
semantics of color::blend(blending_color, fade_rate) changed since 1.5.0
https://github.com/cnjinhao/nana/commit/5c28e72320212f300c74c4205b174e958fcec666#commitcomment-23910500

cnjinhao committed:
Add fontconfig

Add fontconfig, but I don't know why travis-ci was successful with 1.5.4 before https://travis-ci.org/cnjinhao/nana/jobs/270165744

cnjinhao committed:
Add fontconfig

Add fontconfig, but I don't know why travis-ci was successful with 1.5.4 before https://travis-ci.org/cnjinhao/nana/jobs/270165744

cnjinhao committed:
Add fontconfig

Add fontconfig, but I don't know why travis-ci was successful with 1.5.4 before https://travis-ci.org/cnjinhao/nana/jobs/270165744

qPCR4vir committed:
no need to use Wait()

qPCR4vir committed:
delete example project to have uniformity in the solution

qPCR4vir committed:
how ro use set(CMAKE_DEBUG_POSTFIX "_d") ??

qPCR4vir committed:
Merge branch 'hotfix-1.5_with_nana' into hotfix-1.5

qPCR4vir committed:
Merge branch 'develop_with_nana' into develop

qPCR4vir committed:
Merge remote-tracking branch 'origin/develop' into develop

pavelxdd committed:
fix compilation with boost::filesystem

CMakeLists.txt:
---------------
CMake exported definitions with wrong name prefixes when compiling
with Boost Filesystem.

Added boost system and thread libs to find_package, this comment
from cmake documentation doesn't seem to be always true for all
versions of CMake and Boost, also Travis build failed with the
message that boost_system wasn't specified in linker flags, and
threading demo failed due to missing linking to boost_thread lib.
Chrono is needed for thread_pool demo.

demo programs:
--------------
boost::filesystem::path doesn't have a generic_u8string method,
so changed it to nana::to_utf8(generic_wstring).

boost::filesystem::directory_iterator doesn't have begin and end
methods, so changed for loops to make them work with boost.

fixed crashes on accessing some system paths:
Boost can throw an exception when accessing some system paths,
like "C:\Config.Msi", so moved directory_iterator constructors to
try-catch blocks.

cnjinhao committed:
Add boost.system to GCC4.9 linker options

cnjinhao committed:
Add boost.system

It seems the travis cmake doesn't implicitly detect boost.system

cnjinhao committed:
Update CMakeLists.txt

cnjinhao committed:
Update .travis.yml

cnjinhao committed:
Update CMakeLists.txt

cnjinhao committed:
Update CMakeLists.txt

cnjinhao committed:
Update CMakeLists.txt

cnjinhao committed:
Update CMakeLists.txt

qPCR4vir committed:
Merge pull request #13 from pavelxdd/boostfix

fix compilation with boost::filesystem

qPCR4vir committed:
Merge remote-tracking branch 'origin/develop' into hotfix-1.5

# Conflicts:
#	.travis.yml
#	CMakeLists.txt

qPCR4vir committed:
Merge branch 'hotfix-1.5' into develop

qPCR4vir committed:
fix range for

qPCR4vir committed:
boost::filesystem::path doesn't have a generic_u8string method,

qPCR4vir committed:
more boost::filesystem::path doesn't have a generic_u8string method,

qPCR4vir committed:
fix travis - it seemd that a bin already exists from 20170901

qPCR4vir committed:
Merge branch 'develop' into hotfix-1.5 linked to hotfix-1.5.6

cnjinhao committed:
fix the translation string.

qPCR4vir committed:
nana-demo hotfix point to nana hotfix-1.6.1

qPCR4vir committed:
Merge branch 'hotfix'

# Conflicts:
#	.travis.yml
#	CMakeLists.txt

qPCR4vir committed:
message CMAKE_CXX_COMPILER_VERSION

qPCR4vir committed:
more nana paths?

qPCR4vir committed:
VERSION

qPCR4vir committed:
CLion have install

qPCR4vir committed:
travis gcc 8

qPCR4vir committed:
travis cmake 3.12 ?

qPCR4vir committed:
travis cmake 3.12 ? ubuntu install

qPCR4vir committed:
fix? just upgrade version

qPCR4vir committed:
git clone nana branch=CMake

qPCR4vir committed:
git clone branch=CMake https://github.com/qPCR4vir/nana

qPCR4vir committed:
doc

qPCR4vir committed:
simplify cmake usage with include nana

qPCR4vir committed:
Merge remote-tracking branch 'origin/cmake-dev' into cmake-dev

# Conflicts:
#	CMakeLists.txt

qPCR4vir committed:
use add_subdirectory(../nana ../cmake-nana-build-${CONFIG} )

qPCR4vir committed:
fix cd .. in travis ?

qPCR4vir committed:
fix use cmake-dev travis ?

qPCR4vir committed:
build in nana?

qPCR4vir committed:
fix: install to nana-test

qPCR4vir committed:
more animate-bmp

qPCR4vir committed:
trying to fix a definition duplication of boolean window.h vs jpeg

qPCR4vir committed:
more info? not really... ?

qPCR4vir committed:
travis cd ../nana-test and jpeg tested

qPCR4vir committed:
travis cd bin

cnjinhao committed:
debug

cnjinhao committed:
debug

cnjinhao committed:
debug

Why did stdout have no output?

cnjinhao committed:
restore

cnjinhao committed:
restore

cnjinhao committed:
debug

qPCR4vir committed:
reuse NANA_CMAKE_VERBOSE_PREPROCESSOR

qPCR4vir committed:
fix build dir name ?

qPCR4vir committed:
Merge remote-tracking branch 'origin/master' into cmake-dev

# Conflicts:
#	.travis.yml
#	CMakeLists.txt

Commit:
Specify /source-charset:utf-8 option

qPCR4vir committed:
git clone branch=dev from qPCR4vir/nana

cnjinhao committed:
use std::filesystem::path instead of std::string

caused by breaking changes of nana 1.7

cnjinhao committed:
a small fix

cnjinhao committed:
a small fix

cnjinhao committed:
update filebox-txt.cpp

because of breaking changes of nana 1.7

Commit:
fix add - libxcursor-dev in travis

Commit:
Merge remote-tracking branch 'origin/cmake-dev' into cmake-dev

qPCR4vir committed:
Adapted to nana v1.7 - background-effects: crash by window-resize sometimes
compiled all OK.
run:
 - background-effects: crash by window-resize sometimes

qPCR4vir committed:
nana-demo cmake-dev point to qPCR4vir/nana dev. No BOOST. GCC8 only

cnjinhao commented:
Hi,@qPCR4vir I planned to delete this repo, because it always increase our work loading and mislead users. We can maintain the repo of you.

qPCR4vir committed:
update readme

qPCR4vir committed:
clone branch=hotfix-1.7 from cnjinhao/nana

qPCR4vir committed:
clone branch=master from cnjinhao/nana

qPCR4vir committed:
clone branch=develop from cnjinhao/nana

qPCR4vir committed:
Merge pull request #16 from beru/inputbox_source_charset_utf-8

Specify /source-charset:utf-8 option

cnjinhao committed:
treebox disallows the key to be seperatr after 1.7

From 1.7.1, treebox doesn't allow the key to be separators \ or /, and treebox::insert returns empty item proxy if key is / or \

cnjinhao committed:
Changed the parameter of timer.interval

cnjinhao committed:
Changed the parameter type of timer::interval

cnjinhao committed:
Changed the parameter type of timer::interval

qPCR4vir committed:
Merge branch 'hotfix' into develop

# Conflicts:
#	.travis.yml

qPCR4vir committed:
fix min form size

cnjinhao committed:
<nana/gui/wvl.hpp> is removed

cnjinhao committed:
<nana/gui/wvl.hpp> is removed

cnjinhao committed:
<nana/gui/wvl.hpp> is removed

cnjinhao committed:
<nana/gui/wvl.hpp> in favor of <nana/gui.hpp>

qPCR4vir committed:
Merge remote-tracking branch 'origin/develop'

# Conflicts:
#	.travis.yml

qPCR4vir committed:
Merge branch 'master' into hotfix

qPCR4vir committed:
not wanting to start xfvb

qPCR4vir committed:
similar to nana

qPCR4vir committed:
revert generic_u8string to generic_string()

qPCR4vir committed:
using nana::filesystem - todo review!!

qPCR4vir committed:
Merge branch 'develop'

# Conflicts:
#	.travis.yml

qPCR4vir committed:
Merge branch 'hotfix'

# Conflicts:
#	.travis.yml

qPCR4vir committed:
VC2019

qPCR4vir committed:
update travis

qPCR4vir committed:
deprecate example

qPCR4vir committed:
code review

qPCR4vir committed:
Merge remote-tracking branch 'origin/master'

qPCR4vir committed:
nana:: fix

qPCR4vir committed:
add all test to cmake to run using ctest in build directory

qPCR4vir committed:
travis test with ctest?

qPCR4vir committed:
back to run "manually" - better output

qPCR4vir committed:
set working dir

qPCR4vir committed:
fix LanguageStandard>stdcpp17 in all vc2019 projects

qPCR4vir committed:
experiment with cmake and VS2019

qPCR4vir committed:
use std::this_thread::sleep_for in nana 1.8

cnjinhao committed:
Update stretch_image.cpp

fix conflict between nana::drawing and widget::drawing
